contrib/netperf/init.d-collectd-td-deb

```diff
@@ -23,25 +23,52 @@ pidfile="/var/run/${servicename}.pid"
 respawn_pidfile="/var/run/respawn-${servicename}.pid"
 cgroup_root="/sys/fs/cgroup"
 cgroup_path="daemons/netperf"
+cgroup_subpaths="daemons daemons/netperf"
+declare -a cgroup_groups
 
 . /lib/lsb/init-functions
 
+cgroup_has_variable() {
+  ! cgget -r $1 / 2>&1 |
+    fgrep -e 'No such file or directory' -e 'cannot find controller' > /dev/null
+}
+
 # Ubuntu Trusty removes cgconfig for configuring CGroups but does not
 # furnish a replacement. cgroup-lite only provides the CGroups filesystem.
 # Build our configuration directly on that filesystem.  This might break
 # when Debian and/or Ubuntu do provide a replacement.
 configure_cgroup() {
-  local cpu=$cgroup_root/cpu/$cgroup_path
-  local mem=$cgroup_root/memory/$cgroup_path
-  mkdir -p $cpu || log_daemon_msg "Failed to create cgroup $cpu"
-  # TODO(arielshaqed): If Debian doesn't have cgroup memory controller, only
-  # do this when $cgroup_root/memory is loaded.
-  mkdir -p $mem || log_daemon_msg "Failed to create cgroup $mem"
-  echo 1000000 > $cpu/cpu.cfs_period_us
-  echo 15000 > $cpu/cpu.cfs_quota_us
-  echo 180m > $mem/memory.limit_in_bytes
-  # memsw.limit_in_bytes does not exist in Ubuntu Trusty (14.04)
-  echo 0 > $mem/memory.oom_control
+  cgroup_groups[0]=cpu
+  cgcreate -g cpu:$cgroup_path
+  
+  (
+    cgroup_has_variable cpu.cfs_period_us &&
+    cgset -r cpu.cfs_period_us=1000000 $cgroup_path &&
+    cgset -r cpu.cfs_quota_us=15000 $cgroup_path
+  ) || (
+    # E.g. Debian does not support quota and period, use cpu.shares
```

Comment:
Actually it's quite important! We first try to use CFS control over CPU usage, which does what we want. But Debian doesn't come with CFS scheduler out of the box (AFAICT) so that doesn't work. The `||` switches us to the alternative, which sets `cpu.shares` on the cgroup.